### PR TITLE
docs: Document launcher_utils and Fluent launcher modes

### PR DESCRIPTION
**Pull request by hpohekar**
Closes https://github.com/ansys/pyfluent/issues/2518 and https://github.com/ansys/pyfluent/issues/2537

1. Documented launcher_utils.py
2. Documented container, pim and standalone launchers

image

`-------------------------------------------------------------------`

image

`-------------------------------------------------------------------`

image

`-------------------------------------------------------------------`

image

`-------------------------------------------------------------------`

image

### CODE REVIEW

File: doc/source/api/general/launcher/container_launcher/index.rst [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+Container launchers
+===================
+
+==============================  ============================================================
+:ref:`ref_launcher_container`   | Launch Fluent in container mode using ``launch_fluent()``.
+
+:ref:`ref_container_launcher`   Launch Fluent in container mode using ``create_launcher()``.
+==============================  ============================================================
+
```
Comment (hpohekar) [inline doc/source/api/general/launcher/container_launcher/index.rst:11]: @seanpearsonuk @mkundu1 

What should be title and description for these 2 container launch functions ?

File: src/ansys/fluent/core/launcher/container_launcher.py [hunk 2]
```diff
@@ -1,4 +1,16 @@
...
+--------
+
+>>> from ansys.fluent.core.launcher.launcher import create_launcher
+
+>>> container_meshing_launcher = create_launcher("container", mode="meshing")
+>>> container_meshing_session = container_meshing_launcher()
+
+>>> container_solver_launcher = create_launcher("container")
+>>> container_solver_session = container_solver_launcher()
+"""
```
Comment (hpohekar) [inline src/ansys/fluent/core/launcher/container_launcher.py:13]: `create_launcher()` method was mentioned in `Fluent launcher` but it's examples were missing.

Comment (hpohekar) [reply src/ansys/fluent/core/launcher/container_launcher.py:13]: image

File: doc/source/api/general/launcher/container_launcher/index.rst (hunk 1 above)
Comment (seanpearsonuk) [reply doc/source/api/general/launcher/container_launcher/index.rst:11]: The second one is a factory function that creates a callable launcher.

Comment (hpohekar) [reply doc/source/api/general/launcher/container_launcher/index.rst]: It's updated now.

image

**Review by prmukherj — APPROVED** [review]

File: src/ansys/fluent/core/launcher/pim_launcher.py [hunk 3]
```diff
@@ -1,4 +1,16 @@
...
+
+Examples
+--------
+
+>>> from ansys.fluent.core.launcher.launcher import create_launcher
+
+>>> pim_meshing_launcher = create_launcher("pim", mode="meshing")
+>>> pim_meshing_session = pim_meshing_launcher()
+
+>>> pim_solver_launcher = create_launcher("pim")
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/launcher/pim_launcher.py:11]: Do we still have the enum? I think we should support it through this interface.

Comment (hpohekar) [reply src/ansys/fluent/core/launcher/pim_launcher.py]: Yes, done. Thank you.

**Review by mkundu1 — APPROVED** [review]

**Review by seanpearsonuk — APPROVED** [review]

**Pull request merged**